backend/wallets/handlers/wallet.py

```diff
@@ -0,0 +1,94 @@
+
+from django.http import Http404
+
+from rest_framework.response import Response
+from rest_framework.request import Request
+from rest_framework.views import APIView
+
+from wallets.controllers import WalletController
+from wallets.serializers.wallet import WalletCreateSerializer, WalletSerializer, WalletListParametersSerializer
+from wallets.permissions import IsOwnerOrCoOwner, IsOwner
+
+
+class WalletView(APIView):
+    """
+    Handle the GET and POST requests for the Wallet model.
+    """
+
+    def get(self, request: Request) -> Response:
+            
+            request_parameters_serializer = WalletListParametersSerializer(data=request.query_params)
+            request_parameters_serializer.is_valid(raise_exception=True)
+
+            wallets = WalletController.list_wallets(request)
+            return Response(WalletSerializer(wallets, many=True).data, status=200)
+    
+    def post(self, request: Request) -> Response:
+            
+            serializer = WalletCreateSerializer(data=request.data, context={'request': request})
+    
+            if serializer.is_valid():
+    
+                wallet = WalletController.build_wallet(request.data, request.user)
+                return Response(WalletCreateSerializer(wallet).data, status=201)
+            
+            return Response(serializer.errors, status=400)
```

Comment:
See how it looks much more organized when independent? You will also see how it makes your life easier when testing.